### ISSUE TITLE
Cannot add a download without a populated file list 

### ISSUE THREAD

**Issue by drew2a** [Tribler/tribler#7768]
**Describe the bug**

Downloads do not appear on the Downloads page after clicking the "download" button.

**To Reproduce**
1. Run Tribler
2. Search for "ubuntu"
3. Press the download icon for a torrent with 0 or just a few seeders
4. Quickly hit the "Download" button before the file list has a chance to populate
5. Go to the Downloads Page

**Expected behavior**
I expect this download on the Downloads Page, but it is empty.
<img width="870" alt="image" src="https://github.com/Tribler/tribler/assets/13798583/2c6639c3-369e-4859-b064-24125c4c9654">

<img width="798" alt="image" src="https://github.com/Tribler/tribler/assets/13798583/cb89ff55-4cd7-4488-bccc-a34d6daf1796">

**Desktop (please complete the following information):**

- OS: macOS
- Tribler's version: main `a25df64ad67e263fe84f65a0faa4753737bf914b`
- Python: 3.9.6

**Comment by qstokkink** [comment]
Cannot reproduce on `Ubuntu 20.04.6 LTS`/`Python 3.8.10`.

**Comment by drew2a** [comment]
Update: If you wait until the file list appears and then press "Download", it works. If you press download immediately after the dialog shows, then the download is not added.

**Comment by drew2a** [comment]
I guess it is easier to reproduce the bug if you choose a torrent with just a few seeders.

**Comment by drew2a** [comment]
Somewhy the new download went to this execution path:

```
[tribler-core PID:91107] 2023-12-11 12:58:40,060 - INFO - DownloadManager(549) - Start download from URI: magnet:?xt=urn:btih:b22ec09e5c45ad737f18010228c24f7acee4d3a6&dn=Ubuntu 16.10
[tribler-core PID:91107] 2023-12-11 12:58:40,060 - INFO - DownloadManager(557) - Magnet scheme detected
[tribler-core PID:91107] 2023-12-11 12:58:40,061 - INFO - DownloadManager(559) - Name: Ubuntu 16.10. Infohash: b'\xb2.\xc0\x9e\\E\xads\x7f\x18\x01\x02(\xc2Oz\xce\xe4\xd3\xa6'
[tribler-core PID:91107] 2023-12-11 12:58:40,061 - INFO - DownloadManager(566) - Metainfo not found in cache
[tribler-core PID:91107] 2023-12-11 12:58:40,062 - INFO - DownloadManager(577) - Starting download: filename: None, torrent def: <tribler.core.components.libtorrent.torrentdef.TorrentDefNoMetainfo object at 0x122605880>
[tribler-core PID:91107] 2023-12-11 12:58:40,062 - INFO - DownloadManager(595) - Infohash: b'\xb2.\xc0\x9e\\E\xads\x7f\x18\x01\x02(\xc2Oz\xce\xe4\xd3\xa6'. Download: Download <name: 'metainfo request' hops: 1 checkpoint_disabled: 1>
[tribler-core PID:91107] 2023-12-11 12:58:40,062 - INFO - DownloadManager(598) - Download exists and metainfo is not requested.
```

and ended up on this return:

https://github.com/Tribler/tribler/blob/a25df64ad67e263fe84f65a0faa4753737bf914b/src/tribler/core/components/libtorrent/download_manager/download_manager.py#L592

The additional logging added here: #7769

**Comment by qstokkink** [comment]
I reproduced this yesterday. After clicking back and forth between the "All" and "Downloading" tabs, the download would appear. This may be a refresh issue.

**Comment by drew2a** [comment]
This behavior was introduced in
- https://github.com/Tribler/tribler/pull/7666

I've tested it with the following commits: 
- https://github.com/Tribler/tribler/commit/d018d66c655bedb177efc2a17bde7dc1c536736b. Works correctly. 
- https://github.com/Tribler/tribler/commit/f610c82ab5ba02004244131e0ebab8d621600c7d. Works incorrectly.

**Comment by kozlovsky** [comment]
Fixed in #7939
